api/v1alpha3/vspherecluster_types.go

```diff
@@ -60,6 +61,9 @@ type VSphereClusterSpec struct {
 // VSphereClusterStatus defines the observed state of VSphereClusterSpec
 type VSphereClusterStatus struct {
 	Ready bool `json:"ready"`
+
+	// FailureDomains is a slice of failure domain objects synced from the infrastructure provider.
```

Comment:
This is the infra provider so this comment needs adjusting

Reply:
/hold

let's leave the PR open until the implementation

Reply:
Curious to know is there any thoughts around the scope that failure domain will, or plan to, cover, like across multiple datacenters ? or across multiple clusters within one datacenter? Will it support different datastores, networks ?

Reply:
@jzhoucliqr do you have any ideas on how you'd expect it to work, and what you'd want it to cover?

Reply:
one of the use case that we have is to across multiple compute clusters within one datacenter, with non-shared datastores and different networks. I guess this might be a common requirement in terms of HA for control plane nodes.